src/otx/core/schedulers/callable.py

```diff
@@ -0,0 +1,160 @@
+            self.scheduler_init = scheduler_cls
+            self.scheduler_path = scheduler_cls.__module__ + "." + scheduler_cls.__qualname__
+        else:
+            raise TypeError(scheduler_cls)
+
+        self.scheduler_kwargs = scheduler_kwargs
+        self.__dict__.update(scheduler_kwargs)
+
+    def __call__(self, optimizer: Optimizer) -> LRScheduler:
+        """Create `torch.optim.LRScheduler` instance for the given parameters."""
+        return self.scheduler_init(optimizer, **self.scheduler_kwargs)
+
+    def to_lazy_instance(self) -> ClassType:
+        """Return lazy instance of this class.
+
+        Because OTX is rely on jsonargparse library,
+        the default value of class initialization
+        argument should be the lazy instance.
+        Please refer to https://jsonargparse.readthedocs.io/en/stable/#default-values
+        for more details.
+
+        Examples:
+            This is an example to implement a new model with a `StepLR` scheduler and
+            custom configurations as a default.
+
+            ```python
+            class MyAwesomeMulticlassClsModel(OTXMulticlassClsModel):
+                def __init__(
+                    self,
+                    num_classes: int,
+                    optimizer: OptimizerCallable = DefaultOptimizerCallable,
+                    scheduler: LRSchedulerCallable | LRSchedulerListCallable = SchedulerCallableSupportHPO(
+                        scheduler_cls=StepLR,
+                        scheduler_kwargs={
+                            "step_size": 10,
+                            "gamma": 0.5,
+                        },
+                    ).to_lazy_instance(),
+                    metric: MetricCallable = MultiClassClsMetricCallable,
+                    torch_compile: bool = False,
+                ) -> None:
+                ...
+            ```
+        """
+        return lazy_instance(
+            SchedulerCallableSupportHPO,
+            scheduler_cls=self.scheduler_path,
+            scheduler_kwargs=self.scheduler_kwargs,
+        )
+
+    def __reduce__(self) -> str | tuple[Any, ...]:
+        return self.__class__, (
+            self.scheduler_path,
+            self.scheduler_kwargs,
+        )
+
+    @classmethod
+    def from_callable(cls, func: LRSchedulerCallable) -> SchedulerCallableSupportHPO:
+        """Create this class instance from an existing optimizer callable."""
+        dummy_params = [nn.Parameter()]
+        optimizer = Optimizer(dummy_params, {"lr": 1.0})
+        scheduler = func(optimizer)
+
+        allow_names = set(inspect.signature(scheduler.__class__).parameters)
+
+        if isinstance(scheduler, ReduceLROnPlateau):
+            # NOTE: Other arguments except "monitor", such as "patience"
+            # are not included in the signature of ReduceLROnPlateau.__init__()
+            allow_names.update(key for key in inspect.signature(TorchReduceLROnPlateau).parameters)
+
+        block_names = {"optimizer", "last_epoch"}
```

Comment:
Why is last_epoch excluded?

Reply:
![image](https://github.com/openvinotoolkit/training_extensions/assets/26541465/c183cc94-36c1-4fd3-b535-ec7f8c8a7ffa)
`scheduler.state_dict()` returns `last_epoch=0` even if it gets `last_epoch=-1` (default) init argument. `last_epoch>=0` has a problem if not resuming an optimizer.